aws/spec/stack.go

```diff
@@ -82,6 +127,119 @@ func (cmd *UpdateStack) ExtractResult(i interface{}) string {
 	return StringValue(i.(*cloudformation.UpdateStackOutput).StackId)
 }
 
+// Add StackFile support via BeforeRun hook
+// https://github.com/wallix/awless/issues/145
+// http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/continuous-delivery-codepipeline-cfn-artifacts.html
+func (cmd *UpdateStack) BeforeRun(ctx map[string]interface{}) error {
+	// don't do anything if StackFile not provided
```

Comment:
Wondering if we can we avoid duplication and do an extract method on the BeforeRun of Update and Create .... if the end result is not too ugly and makes sense ;) ?

Reply:
Make complete sense, I thought about it too, just didn't have time yet, to play with it a bit more.